packages/react/menu/src/Menu.tsx

```diff
@@ -94,65 +237,179 @@ const MenuContent = React.forwardRef((props, forwardedRef) => {
   return (
     <Presence present={forceMount || context.open}>
       <CollectionSlot>
-        <MenuContentImpl
-          data-state={getOpenState(context.open)}
-          {...contentProps}
-          ref={forwardedRef}
-        />
+        {context.isSubmenu ? (
+          <MenuSubContent {...contentProps} ref={forwardedRef} />
+        ) : (
+          <MenuRootContent {...contentProps} ref={forwardedRef} />
+        )}
       </CollectionSlot>
     </Presence>
   );
 }) as MenuContentPrimitive;
 
-type FocusScopeOwnProps = Polymorphic.OwnProps<typeof FocusScope>;
-type DismissableLayerOwnProps = Polymorphic.OwnProps<typeof DismissableLayer>;
-type RovingFocusGroupOwnProps = Polymorphic.OwnProps<typeof RovingFocusGroup>;
+/* ---------------------------------------------------------------------------------------------- */
 
-type MenuContentImplOwnProps = Polymorphic.Merge<
-  Polymorphic.OwnProps<typeof PopperPrimitive.Content>,
-  Omit<DismissableLayerOwnProps, 'onDismiss'> & {
-    /**
-     * Whether focus should be trapped within the `MenuContent`
-     * (default: false)
-     */
-    trapFocus?: FocusScopeOwnProps['trapped'];
+type MenuRootContentOwnProps = Omit<
+  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  keyof MenuContentImplPrivateProps
```

Comment:
I like this pattern, makes me think we can probably do that in other places 👍

Reply:
Yeah, I was wondering that too...👀 👍